app/Models/Bible/Bible.php

```diff
@@ -397,4 +399,23 @@ public function scopeIsContentAvailable($query, $key)
             [$key]
         );
     }
+    public function scopeIsTimingInformationAvailable($query)
+    {
+
+        $bibles_ids_with_timestamps = Bible::select('bibles.id')
```

Comment:
this doesn't look very efficient.  What I was thinking was that the original query would contain a join to bible_file_timestamps to eliminate bibles. 
"distinct" should not be needed, and is expensive

Reply:
@bradflood I have tested the main query with the join to `bible_file_timestamps` and ti increased the time around 1.5 seconds. The case is that we need the entities: `bible_files` and `bible_fileset_connections` to join the `bible_file_timestamps`. I have tested it with a join and with the exists operator and the time response was the same. Also, we can't join the entity  `bible_file_timestamps` without before to reduce the scope.

Let me share with you an example about the main query with the  join to bible_file_timestamps. The example take around 800ms and this query should execute twice because the pagination should count the total of records:

```sql
SELECT
MIN(current_title.name) AS ctitle,
MIN(ver_title.name) AS vtitle,
MIN(bibles.language_id) AS language_id,
MIN(languages.iso) AS iso,
MIN(bibles.DATE) AS DATE,
MIN(language_autonym.name) AS language_autonym,
MIN(language_current.name) AS language_current,
MIN(bibles.priority) AS priority,
MIN(bibles.id) AS id
FROM (
    SELECT DISTINCT `bibles`.`id`, `bibles`.`language_id`, `bibles`.DATE, `bibles`.priority
    FROM `bibles`
    INNER JOIN `bible_fileset_connections` ON `bibles`.`id` = `bible_fileset_connections`.`bible_id`
    INNER JOIN `bible_files` ON `bible_files`.`hash_id` = `bible_fileset_connections`.`hash_id`
    INNER JOIN (SELECT DISTINCT bible_file_timestamps.bible_file_id FROM bible_file_timestamps) AS timestamps_counts
    ON `timestamps_counts`.`bible_file_id` = `bible_files`.`id`
) AS `bibles`
LEFT JOIN `bible_translations` AS `ver_title` ON `ver_title`.`bible_id` = `bibles`.`id` and `ver_title`.`vernacular` = 1
LEFT JOIN `bible_translations` AS `current_title` ON `current_title`.`bible_id` = `bibles`.`id` and `current_title`.`language_id` = 6414
LEFT JOIN `languages` AS `languages` ON `languages`.`id` = `bibles`.`language_id`
LEFT JOIN `language_translations` AS `language_autonym` ON `language_autonym`.`language_source_id` = `bibles`.`language_id`
    and `language_autonym`.`language_translation_id` = `bibles`.`language_id`
LEFT JOIN `language_translations` AS `language_current` ON `language_current`.`language_source_id` = `bibles`.`language_id`
    and `language_current`.`language_translation_id` = 6414
WHERE EXISTS (
    SELECT 1
    FROM `bible_filesets` 
    INNER JOIN `bible_fileset_connections` ON `bible_fileset_connections`.`hash_id` = `bible_filesets`.`hash_id`
    WHERE `bibles`.`id` = `bible_fileset_connections`.`bible_id` and EXISTS (
        SELECT 1
        FROM dbp_users_2021_12_14.user_keys uk
        JOIN dbp_users_2021_12_14.access_group_api_keys agak ON agak.key_id = uk.id
        JOIN dbp_2021_12_14.access_group_filesets agf ON agf.access_group_id = agak.access_group_id
        WHERE uk.KEY = '52e62d4c-f7c8-4a8b-9008-8634d0fbddb0' and agf.hash_id = bible_filesets.hash_id
    ) and not EXISTS (
        SELECT * FROM `bible_fileset_tags`
        WHERE `bible_filesets`.`hash_id` = `bible_fileset_tags`.`hash_id` and `bible_fileset_tags`.`description` IS NULL
    )
) GROUP BY `bibles`.`id`;
```

Also, I share you a screenshot about the response time with the join to bible_file_timestamps:
![screenshot-localhost_8080-2021 12 28-22_59_37](https://user-images.githubusercontent.com/73488660/147626978-730f1530-f38c-4001-9d11-fa341cadbcc8.png)

Finally, I share you a screenshot with the current solution:
![screenshot-localhost_8080-2021 12 28-23_13_40](https://user-images.githubusercontent.com/73488660/147626988-83d35ecc-e215-4653-8b3c-22d5b1208747.png)

Reply:
@bradflood let me know if we need to change the current logic to use the join directly.

Reply:
@vichugofsl thank you for this detail. To clarify my understanding, is the total time to execute the bibles endpoint with the audio_timings filter the 752 msec (highlighted above), or is that just the time to execute the audio_timings subquery?

Reply:
@bradflood, for the above example the 752 msec value is the total time to execute the bibles endpoint with the audio_timings. For general, the average time is around less to 1 second. But it should be noted that the audio_timings subquery takes the most of part of total response time because the other queries are executed really faster.